x/storage/keeper/cross_app_executor.go

```diff
@@ -0,0 +1,274 @@
+package keeper
+
+import (
+	"encoding/hex"
+	"fmt"
+	"strings"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/common"
+
+	paymentmoduletypes "github.com/bnb-chain/greenfield/x/payment/types"
+	"github.com/bnb-chain/greenfield/x/storage/types"
+)
+
+var _ sdk.CrossChainApplication = &ExecutorApp{}
+
+type ExecutorApp struct {
+	sKeeper    types.StorageKeeper
+	sMsgServer types.StorageMsgServer
+	pMsgServer types.PaymentMsgServer
+}
+
+func NewExecutorApp(storageKeeper types.StorageKeeper, storageMsgServer types.StorageMsgServer, paymentMsgServer types.PaymentMsgServer) *ExecutorApp {
+	return &ExecutorApp{
+		sKeeper:    storageKeeper,
+		sMsgServer: storageMsgServer,
+		pMsgServer: paymentMsgServer,
+	}
+}
+
+func (app *ExecutorApp) ExecuteSynPackage(ctx sdk.Context, appCtx *sdk.CrossChainAppContext, payload []byte) sdk.ExecuteResult {
+	pack, err := DeserializeSynPackage(payload)
+	if err != nil {
+		app.sKeeper.Logger(ctx).Error("deserialize executor syn package error", "payload", hex.EncodeToString(payload), "error", err.Error())
+		panic("deserialize executor syn package error")
```

Comment:
Panic will case a failAck package which will case gas on bsc/opbnb, is that ok the oracle module do not handle the crash of this channel?

Reply:
Modifying oracle module is not convenient. I add a recover here.

Reply:
This change is meaningless.